plugins/out_s3/s3_store.c

```diff
@@ -223,6 +225,7 @@ static int set_files_context(struct flb_s3 *ctx)
                 continue;
             }
             s3_file->fsf = fsf;
+            s3_file->first_log_time = time(NULL);
```

Comment:
No action needed here, just a note:

If we ever come back to resolving the upload time on init upload, this is the line that is causing the init case to use current time rather than first_log time.